tests/suites/test_suite_psa_crypto_metadata.function

```diff
@@ -629,6 +629,28 @@ void block_cipher_key_type( int type_arg, int block_size_arg )
     TEST_EQUAL( type & PSA_KEY_TYPE_CATEGORY_MASK,
                 PSA_KEY_TYPE_CATEGORY_SYMMETRIC );
     TEST_EQUAL( PSA_BLOCK_CIPHER_BLOCK_LENGTH( type ), block_size );
+
+    /* PSA_ROUND_UP_TO_MULTIPLE(block_size, length) in crypto_sizes.h
+     * Requires block sizes to be a power of 2.
+     * The following creates a bit and shifts along until it finds a
+     * match or a mismatch.
+     */
+    int check = 0;
+
+    for (size_t index = 1; index > 0; index = index << 1)
+    {
+        if (index == block_size)
+        {
+            check = 0;
+            break;
+        }
+        if (index > block_size)
+        {
+            check = 1;
+            break;
+        }
+    }
+    TEST_EQUAL( check, 0);
```

Comment:
This is functionally correct but hard to read and overly complicated.

The check could just be something like

```
/* Check that the block size is a power of 2. This is required, at least, for PSA_ROUND_UP_TO_MULTIPLE(block_size, length) in crypto_sizes.h. */
TEST_ASSERT( ( ( block_size - 1 ) & block_size ) == 0 );
```

Reply:
I probably could have worked that out from you're last suggestion if I had given it more thought. I will try be more critical in finding less complex solutions in the future

Reply:
I have used your suggestion which in the process eliminates the other problems